src/services/exposed/records-getter.js

```diff
@@ -12,6 +15,45 @@ class RecordsGetter extends AbstractRecordService {
         return records;
       });
   }
+
+  // NOTICE: This function accept either query or body (that contains ID list) params
+  //         and return an ID list. It could be used to handle both "select all" (query)
+  //         and "select some" (ids).
+  async getIdsFromRequest(params) {
+    // NOTICE: If it receives a list of ID, return it as is.
+    if (params.body
+      && params.body.data
+      && params.body.data.attributes
+      && params.body.data.attributes.ids) {
+      return params.body.data.attributes.ids;
+    }
+
+    // NOTICE: Build id from primary keys (there can be multiple primary keys).
+    //         See: https://github.com/ForestAdmin/forest-express-sequelize/blob/42283494de77c9cebe96adbe156caa45c86a80fa/src/services/composite-keys-manager.js#L24-L40
+    const { primaryKeys } = Schemas.schemas[this.Implementation.getModelName(this.model)];
+    const getId = (record) => primaryKeys.map((primaryKey) => record[primaryKey]).join('-');
+
+    // NOTICE: Get records count
+    const recordsCount = await new this.Implementation.ResourcesGetter(
+      this.model,
+      this.lianaOptions,
+      params,
+    ).count(params.query);
+
+    // NOTICE: records IDs are returned with a batch.
+    const recordsIds = await Array.from({ length: Math.ceil(recordsCount / BATCH_PAGE_SIZE) })
+      .reduce(async (accumulator, _, index) => [...await accumulator, ...(await this.getAll(
+        { ...params.query, page: { number: `${index + 1}`, size: `${BATCH_PAGE_SIZE}` } },
+      )).map((record) => getId(record))], []);
```

Comment:
Maybe that's just me, but I found this part to be difficult to understand with current indentation :confused:

Reply:
Fixed thanks!